lib/lrmd/lrmd_client.c

```diff
@@ -1452,11 +1498,32 @@ lrmd_tcp_connect_cb(void *userdata, int rc, int sock)
         return;
     }
 
-    if (tls_client_handshake(lrmd) != pcmk_rc_ok) {
-        return;
-    }
+    /* If the TLS handshake immediately succeeds or fails, we can handle that
+     * now without having to deal with mainloops and retries.  Otherwise, add a
+     * trigger to keep trying until we get a result (or it times out).
+     */
+    rc = pcmk__tls_client_try_handshake(native->remote, &tls_rc);
+    if (rc == EAGAIN) {
+        struct handshake_data_s *hs = NULL;
+
+        if (native->handshake_trigger != NULL) {
+            return;
```

Comment:
I suspect this is not possible in normal operation, probably only if the calling code tries to do two async connections to the same executor at the same time.